src/compiler/checker.ts

```diff
@@ -13876,6 +13859,12 @@ namespace ts {
                 case SyntaxKind.AmpersandAmpersandToken:
                 case SyntaxKind.CommaToken:
                     return node === right ? getContextualType(binaryExpression) : undefined;
+                case SyntaxKind.EqualsEqualsEqualsToken:

```

Comment:
this changes what contextual type mean in the language. we should talk about this first